sqlglot/parser.py

```diff
@@ -920,6 +920,8 @@ class Parser(metaclass=_Parser):
     OPCLASS_FOLLOW_KEYWORDS = {"ASC", "DESC", "NULLS"}
     OPTYPE_FOLLOW_TOKENS = {TokenType.COMMA, TokenType.R_PAREN}
 
+    SQL_READ_WRITE_CHARACTERISTICS = {("NO", "SQL"), ("CONTAINS", "SQL"), ("READS", "SQL", "DATA"), ("MODIFIES", "SQL", "DATA")}
```

Comment:
this is a bit inefficient because for every property now, you have to do these extraneous for loop. instead, could you add "NO", "CONTAINS", "READS", "MODIFIES", to the property_parsers, and within there, finish the parsing?